src/www/ui/api/Controllers/UploadTreeController.php

```diff
@@ -138,4 +140,60 @@ public function setClearingDecision($request, $response, $args)
       return $response->withJson($returnVal->getArray(), $returnVal->getCode());
     }
   }
+
+  /**
+   * Get the clearing history for a particular upload-tree
+   *
+   * @param ServerRequestInterface $request
+   * @param ResponseHelper $response
+   * @param array $args
+   * @return ResponseHelper
+   */
+  public function getClearingHistory($request, $response, $args)
+  {
+    $itemId = intval($args['itemId']);
+    $uploadId = intval($args['id']);
+    $uploadDao = $this->restHelper->getUploadDao();
+    $clearingDao = $this->container->get('dao.clearing');
+
+    $returnVal = null;
+    $decisionKeysMap = [
+      3 => "TO_BE_DISCUSSED",
+      4 => "IRRELEVANT",
+      5 => "IDENTIFIED",
+      6 => "DO_NOT_USE",
+      7 => "NON_FUNCTIONAL"
+    ];
```

Comment:
Can the object of class `\Fossology\Lib\Data\DecisionTypes` be used instead and use it's function `getTypeName($type)` to get the display name? It will reduce the duplication of name storage arrays.

Reply:
I agree, but I have a different perspective. I think returning constants is a better option. This way, the UI can map them to the appropriate display texts for each type instead of deciding on the backend what the UI should display.  Also, the UI comparison process will be simpler, if we want to customize the display for a specific type, comparing with the value `TO_BE_DISCUSSED` is easier than when we have `To be discussed` as a value.

Reply:
Let me know what you think about this ? @GMishx